addons/mrp_account/models/stock_rule.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = 'stock.rule'
+
+    def _prepare_mo_vals(self, product_id, product_qty, product_uom, location_id, name, origin, company_id, values, bom):
+        vals = super()._prepare_mo_vals(product_id, product_qty, product_uom, location_id, name, origin, company_id, values, bom)
+        vals.update({'analytic_account_id': bom.analytic_account_id.id})
+        return vals
```

Comment:
This is probably useless. The onchange on the mrp.production with the bom should be a compute and do this work